### PR TITLE
Update maven version in pom.xml to avoid vulnerabilities

### PR DESCRIPTION
**Pull request by deepesh1511**
Updating maven version to latest

### CODE REVIEW

**Comment by zabil** [comment]
@deepesh1511 can you please sign the DCO? Follow instructions here https://github.com/getgauge-contrib/gauge-maven-plugin/pull/85/checks?check_run_id=20747451448

File: pom.xml [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
-    <version>1.6.1</version>
+    <version>1.6.2</version>
     <packaging>maven-plugin</packaging>
     <name>Gauge Maven Plugin</name>
     <url>http://github.com/getgauge/gauge-maven-plugin</url>
     <description>A maven plugin to execute gauge specs in the project</description>
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.version>3.8.1</maven.version>
+        <maven.version>4.0.0-alpha-12</maven.version>
```
Comment (chadlwilson) [inline pom.xml:12]: I don't really understand how this solves the problem but we shouldnt depend on an unreleased alpha version of Maven.

`3.9.6` is the latest Maven release.

**Review by chadlwilson — CHANGES_REQUESTED** [review]

**Comment by chadlwilson** [comment]
The root of the problem here is likely that these Maven dependencies should be in the `provided` scope for a Maven plugin, not compile time dependencies as they currently are. This was they are supplied by Maven and not resolved by the plugin.

For example, this is how the dependency-check maven is set up: https://mvnrepository.com/artifact/org.owasp/dependency-check-maven/9.0.9 or something core like the [maven-gpg-plugin](https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-gpg-plugin/3.1.0).

An alternate/cleaner way to fix this is at https://github.com/getgauge-contrib/gauge-maven-plugin/pull/87 (despite being an org member, I don't seem to have permissions this particular repository)

**Comment by haroon-sheikh** [comment]
The changes have been updated as part of https://github.com/getgauge-contrib/gauge-maven-plugin/pull/87

**Pull request closed without merging**